nncf/quantization/algorithms/weight_compression/lora_correction.py

```diff
@@ -0,0 +1,244 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from pathlib import Path
+from typing import Dict, List, Optional, Tuple
+
+import matplotlib.pyplot as plt
+import pandas as pd
+
+from nncf.common.logging import nncf_logger
+from nncf.common.utils.debug import DEBUG_LOG_DIR
+from nncf.common.utils.debug import is_debug
+from nncf.parameters import CompressWeightsMode
+from nncf.quantization.advanced_parameters import AdvancedLoraCorrectionParameters
+from nncf.quantization.algorithms.weight_compression.activation_stats import process_stats
+from nncf.quantization.algorithms.weight_compression.config import WeightCompressionParameters
+from nncf.quantization.algorithms.weight_compression.weight_lowering import do_dequantization
+from nncf.quantization.algorithms.weight_compression.weight_lowering import do_nf4_fake_quantization_from_norm_weight
+from nncf.tensor import Tensor
+from nncf.tensor import functions as fns
+from nncf.tensor.definitions import TensorDataType
+
+
+class DebugInterface:
+    """
+    Utility class to collect and dump debug information of the Lora Correction algorithm.
+    """
+
+    def __init__(self):
+        self._noise_per_layer = {}
+
+    def add_noises(self, layer_name: str, value: float):
+        self._noise_per_layer[layer_name] = value
+
+    def dump_data(self):
+        if not self._noise_per_layer:
+            return
+        dump_dir = Path(DEBUG_LOG_DIR) / "lora"
+        dump_dir.mkdir(parents=True, exist_ok=True)
+
+        layer_dir = dump_dir / "per_layer"
+        layer_dir.mkdir(parents=True, exist_ok=True)
+
+        df = pd.DataFrame(self._noise_per_layer)
+        losses_path = dump_dir / "noises.csv"
+        nncf_logger.debug("Quantization noise through the correction process is saved to: ", losses_path)
+        df.to_csv(losses_path)
+
+        for name in df.columns:
+            plt.plot(df[name])
+            filename = name.replace("/", "_") + ".jpg"
+            plt.savefig(layer_dir / filename)
+            plt.clf()
+
+        delta = df.iloc[0] - df.iloc[-1]
+        nncf_logger.debug("Is quantization noise reduced for all layers: ", all(delta > 0))
+
+        _, ax = plt.subplots(1)
+        ax.plot(delta)
+        ax.set_xticklabels([])
+        delta_path = dump_dir / "qnoise_change.jpg"
+        nncf_logger.debug("Saving change in quantization noise for each layer to: ", delta_path)
+        plt.savefig(delta_path)
+        plt.clf()
+
+
+class LoraCorrectionAlgorithm:
+    """
+    Contains implementation of LoRA Correction algorithm.
+
+    The method reduces quantization noise after weight compression using low rank adapters.
+    """
+
+    def __init__(self, activations: Dict[str, List[Tensor]], lora_correction_params: AdvancedLoraCorrectionParameters):
```

Comment:
What do you think about explicit definition of algorithm parameters?

The current implementation: 
```
from nncf.quantization.advanced_parameters import AdvancedLoraCorrectionParameters

algo = LoraCorrectionAlgorithm(
    activations=activations,
    lora_correction_params=AdvancedLoraCorrectionParameters(
        rank=8
        n_iters=3
        w_regularization=False
        subset_size=32
        is_int8_adapters=True
    )
)
```
or 
```
algo = LoraCorrectionAlgorithm(
    activations=activations,
    rank=8
    n_iters=3
    w_regularization=False
    subset_size=32
    is_int8_adapters=True
)
```